job/cache.go

```diff
@@ -115,6 +115,18 @@ func (c *MemoryJobCache) Delete(id string) {
 	c.jobs.Lock.Lock()
 	defer c.jobs.Lock.Unlock()
 
+	j := c.jobs.Jobs[id]
+	if j == nil {
+		// TODO
+		//return nil, JobDoesntExistErr
+		return
+	}
+
+	// TODO return error
+	c.jobs.Lock.Unlock()
```

Comment:
@elimisteve What I just experimented with locally is instead throwing this into a goroutine and using the `Disable()` function before anything else. It shouldn't be a problem, as if for some reason it took a little while to remove itself from its parent jobs, it would not run even if the parent jobs were triggered. This became more needed as the locking and unlocking was getting worse locally since I needed to do the same thing for Parent jobs removing themselves from there Dependent jobs.